service/controller/aws/v14/resource/kubeconfig/desired_test.go

```diff
@@ -0,0 +1,147 @@
+package kubeconfig
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+	ktesting "k8s.io/client-go/testing"
+)
+
+const (
+	kubeconfigYaml = `apiVersion: v1
+kind: Config
+clusters:
+- name: giantswarm-w7utg
+  cluster:
+    server: api.giantswarm.io
+    certificate-authority-data: Y2E=
+users:
+- name: giantswarm-w7utg-user
+  user:
+    client-certificate-data: Y3J0
+    client-key-data: a2V5
+contexts:
+- name: giantswarm-w7utg-context
+  context:
+    cluster: giantswarm-w7utg
+    user: giantswarm-w7utg-user
+current-context: giantswarm-w7utg-context
+preferences: {}
+`
+)
+
+func Test_Resource_GetDesiredState(t *testing.T) {
+	tests := []struct {
+		name           string
+		config         *v1alpha1.AWSClusterConfig
+		expectedSecret *corev1.Secret
+		errorMatcher   func(error) bool
+		secretCert     *corev1.Secret
+	}{
+		{
+			name: "case 0: cluster config",
```

Comment:
```suggestion
			name: "case 0: basic match",
```

We usually use basic match for the first passing test case. I'm not sure what cluster config means in this context.

Are there any other cases that need to be handled?

Reply:
Nope, just a history